composer/datasets/in_context_learning_evaluation.py

```diff
@@ -109,12 +198,193 @@ def update_metric(self, metric, batch, output_logits, labels):
+            prompt_string (str): The prompt to prepend to all inputs
+            example_delimiter (str): The delimiter used to separate each example query/answer pair
+            continuation_delimiter (str): The delimiter used to separate each query from its answer
+
+        Returns:
+            dict: Contains the query, the list of encoded potential answer choices, the preamble (prompt + fewshot examples), and
+                the index of the correct answer choice.
+        """
+        examples = []
+        for sample_idx in tqdm(range(len(self.samples))):
+
+            preamble = prompt_string
+            if num_fewshot > 0:
+                fewshot_idxs = _get_fewshot_sample_idxs(len(self.samples), num_fewshot, sample_idx)
+                for fewshot_idx in fewshot_idxs:
+                    query, choices, gold_idx = self.samples[fewshot_idx]['query'], self.samples[fewshot_idx][
+                        'choices'], self.samples[fewshot_idx]['gold']
+                    if len(preamble) > 0:
+                        query = f'{example_delimiter}{query}'
+                    preamble += f'{query}{continuation_delimiter}{choices[gold_idx]}'
+
+            encoded_example = {}
+            query, choices, gold_idx = self.samples[sample_idx]['query'], self.samples[sample_idx][
+                'choices'], self.samples[sample_idx]['gold'],
+            if len(preamble) > 0:
+                query = f'{example_delimiter}{query}'
+
+            choices = [f'{continuation_delimiter}{choice}' for choice in choices]
+            encoded_example['query'] = self.tokenizer(query)
+            encoded_example['choices'] = [self.tokenizer(choice) for choice in choices]
+            encoded_example['preamble'] = self.tokenizer(
+                preamble)  # if the preamble is empty then these will be 0-length lists
+            encoded_example['gold_idx'] = gold_idx
+
+            examples.append(encoded_example)
+
+        return examples
+
+    def __getitem__(self, index):
+        return self.encoded_dataset[index]
+
+    def __len__(self):
+        return len(self.encoded_dataset)
+
+    def collate_fn(self, data):
+        inputs = []
+        continuation_indices = []
+        gold_idxs = []
+        choice_groupings = []
+        for data_pair in data:
+
+            choice_start_idx = len(continuation_indices)
+            preamble, context, choices, gold_idx = (data_pair['preamble'], data_pair['query'], data_pair['choices'],
+                                                    data_pair['gold_idx'])
+
+            for choice in choices:
+                context_enc = preamble['input_ids'] + context['input_ids']
+                continuation_enc = choice['input_ids']
+                inp, continuation_span = _make_padded_input(context_enc, continuation_enc, self.max_seq_len,
+                                                            self.eos_tok_id)
```

Comment:
Related to below, is there a reason we are padding with the `eos_tok_id` rather than the `pad_tok_id`? I guess if it's consistent all the way through that's fine, but it might be good to add a comment about it.

Reply:
The actual token used by the data loader can be the pad token or the eos token, so for clarity I will change the internal naming to pad token. For the purposes of constructing the data loader, the GPTTokenizer's eos_tok_id is the last id in the GPT embedding, the pad token is the index after, and does not appear in the embedding, hence we can't actually pass in the pad_token_id in our tensors